File: RepeatedSubstringPattern/RepeatedSubstringPattern.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
  return s.repeat(2).slice(1, -1).includes(s);
};


/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
  let pattern = "";
  for (let i = 0; i < Math.floor(s.length / 2); i++) {
    pattern += s[i];
    if (pattern.repeat(s.length / pattern.length) === s) return true;
  }
  return false;
};

File: RepeatedSubstringPattern/RepeatedSubstringPattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        return s in (s + s)[1 : -1]


class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        length = len(s)
        for i in range(1, length // 2 + 1):
            if length % i == 0 and s[:i] * (length // i) == s:
                return True
        return False
